pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -119,15 +120,73 @@ func (jm *CronJobController) SyncAll() {
 	glog.V(4).Infof("Found %d groups", len(jobsBySj))
 
 	for _, sj := range sjs {
-		SyncOne(sj, jobsBySj[sj.UID], time.Now(), jm.jobControl, jm.sjControl, jm.podControl, jm.recorder)
+		sj := syncOne(sj, jobsBySj[sj.UID], time.Now(), jm.jobControl, jm.sjControl, jm.podControl, jm.recorder)
+		cleanupFinishedJobs(sj, jobsBySj[sj.UID], jm.jobControl, jm.sjControl, jm.podControl, jm.recorder)
```

Comment:
Can both `syncOne` and `cleanupFinishedJobs` accept pointer to CronJob, that will get rid of the need to return anything from `syncOne`. I see we're not consistent with this along the code, eg. `removeOldestJob` does that, already.